lib/compat/wordpress-6.4/script-loader.php

```diff
@@ -158,9 +158,13 @@ function _gutenberg_get_iframed_editor_assets_6_4() {
 		}
 	}
 
+	// Remove the deprecated `print_emoji_styles` handler.
+	// It avoids breaking style generation with a deprecation message.
+	remove_action( 'wp_print_styles', 'print_emoji_styles' );
 	ob_start();
 	wp_print_styles();
 	$styles = ob_get_clean();
+	add_action( 'wp_print_styles', 'print_emoji_styles' );
```

Comment:
Should we check to see if `remove_action()` is truthy before re-adding the action? I think that would mean we'd only add it back in again if it was originally present 🤔

Reply:
This is a common pattern for temporarily removing actions. We do the same above.

Reply:
Where do we do it above? Sorry if I'm missing something obvious!

I can see `add_filter( 'should_load_block_editor_scripts_and_styles', '__return_false' );` followed by `remove_filter( 'should_load_block_editor_scripts_and_styles', '__return_false' );`. This one is different in a subtle way, as it first removes before then adding. If when this function is called there was never an action registered for `print_emoji_styles` (e.g. it was unregistered elsewhere), then the `add_action` could restore it unexpectedly?

I'm a little unsure of the consequences of making changes here, which is why I'm a little bit cautious about the changes. I don't quite understand the scope of what was introduced in https://github.com/WordPress/wordpress-develop/pull/4824, so am a tiny bit nervous about the fix in this PR, particularly since this PR didn't have any explanation as to why the fix was required.

I'd probably lean more toward reviewing whether https://github.com/WordPress/wordpress-develop/pull/4824 should be reverted, as it introduced a regression in core. That said, I'm very much in favour of pragmatic fixes that unblock merging in Gutenberg, so please don't take my comments here as a blocker to landing this PR if it fixes things 🙂